homeassistant/components/lutron_caseta/device_trigger.py

```diff
@@ -454,61 +414,69 @@ async def async_get_triggers(
     return triggers
 
 
-def _device_model_to_type(device_registry_model: str) -> str:
-    """Convert a lutron_caseta device registry entry model to type."""
-    model_list = device_registry_model.split(" ")
-    device_type = model_list.pop().replace("(", "").replace(")", "")
-    return _lutron_model_to_device_type(" ".join(model_list), device_type)
-
-
-def _lutron_model_to_device_type(model: str, device_type: str) -> str:
-    """Get the mapped type based on the lutron model or type."""
-    return LUTRON_MODEL_TO_TYPE.get(model, device_type)
-
-
 async def async_attach_trigger(
     hass: HomeAssistant,
     config: ConfigType,
     action: TriggerActionType,
     trigger_info: TriggerInfo,
 ) -> CALLBACK_TYPE:
     """Attach a trigger."""
-    device_registry = dr.async_get(hass)
-    if (
-        not (device := device_registry.async_get(config[CONF_DEVICE_ID]))
-        or not device.model
+    device_id = config[CONF_DEVICE_ID]
+    subtype = config[CONF_SUBTYPE]
+    if not (data := get_lutron_data_by_dr_id(hass, device_id)) or not (
+        keypad := data.keypad_data.dr_device_id_to_keypad[device_id]
     ):
         raise HomeAssistantError(
-            f"Cannot attach trigger {config} because device with id {config[CONF_DEVICE_ID]} is missing or invalid"
+            f"Cannot attach trigger {config} because device with id {device_id} is missing or invalid"
         )
-    device_type = _device_model_to_type(device.model)
-    _, serial = list(device.identifiers)[0]
-    schema = DEVICE_TYPE_SCHEMA_MAP[device_type]
-    valid_buttons = DEVICE_TYPE_SUBTYPE_MAP_TO_LEAP[device_type]
+
+    keypad_trigger_schemas = data.keypad_data.trigger_schemas
+    keypad_button_names_to_leap = data.keypad_data.button_names_to_leap
+
+    device_type = keypad["type"]
+    serial = keypad["serial"]
+
+    # Retrieve trigger schema, preferring hard-coded triggers from device_trigger.py
+    schema = DEVICE_TYPE_SCHEMA_MAP.get(
+        device_type,
+        keypad_trigger_schemas.get(keypad["lutron_device_id"]),
```

Comment:
```suggestion
        device_type, keypad_trigger_schemas[lutron_device_id]
```

We shouldn't use `.get` here for the final attempt since if its missing it would be a bug and then `schema` would be `None`